Install issues on Redhat Enterprise v 7

After stepping through the process as documented on a brand new Redhat VM I hit a problem where I cannot find pbr. 

makefile: Target install done.
Error: Package version could not be determined (requires pbr; run "make install")
make: *** [_check_version] Error 1
(pywbem27)kschopmeyer@smipypingtest3 pywbem [master] $ make install
makefile: Target install done.
(pywbem27)kschopmeyer@smipypingtest3 pywbem [master] $ pip list
Package    Version      Location 


 I am creating this issue now to capture the data before I play around. See second comment for results after trying again

Here is the last part of the process after I had done the  ./pywbem_os_setup.sh

--> Processing Dependency: libsepol-devel(x86-64) >= 2.5-6 for package: libselinux-devel-2.5-12.el7.x86_64
--> Processing Dependency: pkgconfig(libpcre) for package: libselinux-devel-2.5-12.el7.x86_64
--> Processing Dependency: pkgconfig(libsepol) for package: libselinux-devel-2.5-12.el7.x86_64
---> Package libverto-devel.x86_64 0:0.2.5-4.el7 will be installed
--> Running transaction check
---> Package libsepol-devel.x86_64 0:2.5-8.1.el7 will be installed
---> Package pcre-devel.x86_64 0:8.32-17.el7 will be installed
--> Finished Dependency Resolution

Dependencies Resolved
    
''
================================================================================
   Package               Arch     Version              Repository            Size
  ================================================================================
Installing:
 openssl-devel         x86_64   1:1.0.2k-12.el7      rhel-7-server-rpms   1.5 M
Installing for dependencies:
 keyutils-libs-devel   x86_64   1.5.8-3.el7          rhel-7-server-rpms    37 k
 krb5-devel            x86_64   1.15.1-19.el7        rhel-7-server-rpms   269 k
 libcom_err-devel      x86_64   1.42.9-12.el7_5      rhel-7-server-rpms    31 k
 libkadm5              x86_64   1.15.1-19.el7        rhel-7-server-rpms   175 k
 libselinux-devel      x86_64   2.5-12.el7           rhel-7-server-rpms   186 k
 libsepol-devel        x86_64   2.5-8.1.el7          rhel-7-server-rpms    77 k
 libverto-devel        x86_64   0.2.5-4.el7          rhel-7-server-rpms    12 k
 pcre-devel            x86_64   8.32-17.el7          rhel-7-server-rpms   480 k
 zlib-devel            x86_64   1.2.7-17.el7         rhel-7-server-rpms    50 k

Transaction Summary
================================================================================
Install  1 Package (+9 Dependent packages)

Total download size: 2.8 M
Installed size: 6.0 M
Downloading packages:
(1/10): keyutils-libs-devel-1.5.8-3.el7.x86_64.rpm         |  37 kB   00:00     
(2/10): krb5-devel-1.15.1-19.el7.x86_64.rpm                | 269 kB   00:00     
(3/10): libcom_err-devel-1.42.9-12.el7_5.x86_64.rpm        |  31 kB   00:00     
(4/10): libkadm5-1.15.1-19.el7.x86_64.rpm                  | 175 kB   00:00     
(5/10): libselinux-devel-2.5-12.el7.x86_64.rpm             | 186 kB   00:00     
(6/10): libsepol-devel-2.5-8.1.el7.x86_64.rpm              |  77 kB   00:00     
(7/10): libverto-devel-0.2.5-4.el7.x86_64.rpm              |  12 kB   00:00     
(8/10): pcre-devel-8.32-17.el7.x86_64.rpm                  | 480 kB   00:00     
(9/10): openssl-devel-1.0.2k-12.el7.x86_64.rpm             | 1.5 MB   00:01     
(10/10): zlib-devel-1.2.7-17.el7.x86_64.rpm                |  50 kB   00:00     
--------------------------------------------------------------------------------
Total                                              891 kB/s | 2.8 MB  00:03     
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : keyutils-libs-devel-1.5.8-3.el7.x86_64                      1/10 
  Installing : libcom_err-devel-1.42.9-12.el7_5.x86_64                     2/10 
  Installing : libkadm5-1.15.1-19.el7.x86_64                               3/10 
  Installing : libsepol-devel-2.5-8.1.el7.x86_64                           4/10 
  Installing : pcre-devel-8.32-17.el7.x86_64                               5/10 
  Installing : libselinux-devel-2.5-12.el7.x86_64                          6/10 
  Installing : libverto-devel-0.2.5-4.el7.x86_64                           7/10 
  Installing : krb5-devel-1.15.1-19.el7.x86_64                             8/10 
  Installing : zlib-devel-1.2.7-17.el7.x86_64                              9/10 
  Installing : 1:openssl-devel-1.0.2k-12.el7.x86_64                       10/10 
  Verifying  : krb5-devel-1.15.1-19.el7.x86_64                             1/10 
  Verifying  : zlib-devel-1.2.7-17.el7.x86_64                              2/10 
  Verifying  : 1:openssl-devel-1.0.2k-12.el7.x86_64                        3/10 
  Verifying  : libverto-devel-0.2.5-4.el7.x86_64                           4/10 
  Verifying  : libselinux-devel-2.5-12.el7.x86_64                          5/10 
  Verifying  : pcre-devel-8.32-17.el7.x86_64                               6/10 
  Verifying  : libsepol-devel-2.5-8.1.el7.x86_64                           7/10 
  Verifying  : libkadm5-1.15.1-19.el7.x86_64                               8/10 
  Verifying  : libcom_err-devel-1.42.9-12.el7_5.x86_64                     9/10 
  Verifying  : keyutils-libs-devel-1.5.8-3.el7.x86_64                     10/10 

Installed:
  openssl-devel.x86_64 1:1.0.2k-12.el7                                          

Dependency Installed:
  keyutils-libs-devel.x86_64 0:1.5.8-3.el7  krb5-devel.x86_64 0:1.15.1-19.el7  
  libcom_err-devel.x86_64 0:1.42.9-12.el7_5 libkadm5.x86_64 0:1.15.1-19.el7    
  libselinux-devel.x86_64 0:2.5-12.el7      libsepol-devel.x86_64 0:2.5-8.1.el7
  libverto-devel.x86_64 0:0.2.5-4.el7       pcre-devel.x86_64 0:8.32-17.el7    
  zlib-devel.x86_64 0:1.2.7-17.el7         

Complete!
pywbem_os_setup.sh: Installing package: gcc-c++
Loaded plugins: product-id, search-disabled-repos, subscription-manager
Resolving Dependencies
--> Running transaction check
---> Package gcc-c++.x86_64 0:4.8.5-28.el7_5.1 will be installed
--> Processing Dependency: gcc = 4.8.5-28.el7_5.1 for package: gcc-c++-4.8.5-28.el7_5.1.x86_64
--> Processing Dependency: libstdc++-devel = 4.8.5-28.el7_5.1 for package: gcc-c++-4.8.5-28.el7_5.1.x86_64
--> Processing Dependency: libmpc.so.3()(64bit) for package: gcc-c++-4.8.5-28.el7_5.1.x86_64
--> Processing Dependency: libmpfr.so.4()(64bit) for package: gcc-c++-4.8.5-28.el7_5.1.x86_64
--> Running transaction check
---> Package gcc.x86_64 0:4.8.5-28.el7_5.1 will be installed
--> Processing Dependency: cpp = 4.8.5-28.el7_5.1 for package: gcc-4.8.5-28.el7_5.1.x86_64
--> Processing Dependency: glibc-devel >= 2.2.90-12 for package: gcc-4.8.5-28.el7_5.1.x86_64
---> Package libmpc.x86_64 0:1.0.1-3.el7 will be installed
---> Package libstdc++-devel.x86_64 0:4.8.5-28.el7_5.1 will be installed
---> Package mpfr.x86_64 0:3.1.1-4.el7 will be installed
--> Running transaction check
---> Package cpp.x86_64 0:4.8.5-28.el7_5.1 will be installed
---> Package glibc-devel.x86_64 0:2.17-222.el7 will be installed
--> Processing Dependency: glibc-headers = 2.17-222.el7 for package: glibc-devel-2.17-222.el7.x86_64
--> Processing Dependency: glibc-headers for package: glibc-devel-2.17-222.el7.x86_64
--> Running transaction check
---> Package glibc-headers.x86_64 0:2.17-222.el7 will be installed
--> Processing Dependency: kernel-headers >= 2.2.1 for package: glibc-headers-2.17-222.el7.x86_64
--> Processing Dependency: kernel-headers for package: glibc-headers-2.17-222.el7.x86_64
--> Running transaction check
---> Package kernel-headers.x86_64 0:3.10.0-862.3.2.el7 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package            Arch      Version               Repository             Size
================================================================================
Installing:
 gcc-c++            x86_64    4.8.5-28.el7_5.1      rhel-7-server-rpms    7.2 M
Installing for dependencies:
 cpp                x86_64    4.8.5-28.el7_5.1      rhel-7-server-rpms    5.9 M
 gcc                x86_64    4.8.5-28.el7_5.1      rhel-7-server-rpms     16 M
 glibc-devel        x86_64    2.17-222.el7          rhel-7-server-rpms    1.1 M
 glibc-headers      x86_64    2.17-222.el7          rhel-7-server-rpms    678 k
 kernel-headers     x86_64    3.10.0-862.3.2.el7    rhel-7-server-rpms    7.1 M
 libmpc             x86_64    1.0.1-3.el7           rhel-7-server-rpms     51 k
 libstdc++-devel    x86_64    4.8.5-28.el7_5.1      rhel-7-server-rpms    1.5 M
 mpfr               x86_64    3.1.1-4.el7           rhel-7-server-rpms    203 k

Transaction Summary
================================================================================
Install  1 Package (+8 Dependent packages)

Total download size: 40 M
Installed size: 84 M
Downloading packages:
(1/9): cpp-4.8.5-28.el7_5.1.x86_64.rpm                     | 5.9 MB   00:06     
(2/9): gcc-c++-4.8.5-28.el7_5.1.x86_64.rpm                 | 7.2 MB   00:05     
(3/9): glibc-devel-2.17-222.el7.x86_64.rpm                 | 1.1 MB   00:01     
(4/9): glibc-headers-2.17-222.el7.x86_64.rpm               | 678 kB   00:00     
(5/9): kernel-headers-3.10.0-862.3.2.el7.x86_64.rpm        | 7.1 MB   00:04     
(6/9): libmpc-1.0.1-3.el7.x86_64.rpm                       |  51 kB   00:00     
(7/9): gcc-4.8.5-28.el7_5.1.x86_64.rpm                     |  16 MB   00:20     
(8/9): mpfr-3.1.1-4.el7.x86_64.rpm                         | 203 kB   00:00     
(9/9): libstdc++-devel-4.8.5-28.el7_5.1.x86_64.rpm         | 1.5 MB   00:01     
--------------------------------------------------------------------------------
Total                                              1.9 MB/s |  40 MB  00:21     
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : mpfr-3.1.1-4.el7.x86_64                                      1/9 
  Installing : libmpc-1.0.1-3.el7.x86_64                                    2/9 
  Installing : cpp-4.8.5-28.el7_5.1.x86_64                                  3/9 
  Installing : kernel-headers-3.10.0-862.3.2.el7.x86_64                     4/9 
  Installing : glibc-headers-2.17-222.el7.x86_64                            5/9 
  Installing : glibc-devel-2.17-222.el7.x86_64                              6/9 
  Installing : gcc-4.8.5-28.el7_5.1.x86_64                                  7/9 
  Installing : libstdc++-devel-4.8.5-28.el7_5.1.x86_64                      8/9 
  Installing : gcc-c++-4.8.5-28.el7_5.1.x86_64                              9/9 
  Verifying  : libstdc++-devel-4.8.5-28.el7_5.1.x86_64                      1/9 
  Verifying  : gcc-4.8.5-28.el7_5.1.x86_64                                  2/9 
  Verifying  : mpfr-3.1.1-4.el7.x86_64                                      3/9 
  Verifying  : glibc-devel-2.17-222.el7.x86_64                              4/9 
  Verifying  : cpp-4.8.5-28.el7_5.1.x86_64                                  5/9 
  Verifying  : glibc-headers-2.17-222.el7.x86_64                            6/9 
  Verifying  : libmpc-1.0.1-3.el7.x86_64                                    7/9 
  Verifying  : gcc-c++-4.8.5-28.el7_5.1.x86_64                              8/9 
  Verifying  : kernel-headers-3.10.0-862.3.2.el7.x86_64                     9/9 

Installed:
  gcc-c++.x86_64 0:4.8.5-28.el7_5.1                                             

Dependency Installed:
  cpp.x86_64 0:4.8.5-28.el7_5.1                                                 
  gcc.x86_64 0:4.8.5-28.el7_5.1                                                 
  glibc-devel.x86_64 0:2.17-222.el7                                             
  glibc-headers.x86_64 0:2.17-222.el7                                           
  kernel-headers.x86_64 0:3.10.0-862.3.2.el7                                    
  libmpc.x86_64 0:1.0.1-3.el7                                                   
  libstdc++-devel.x86_64 0:4.8.5-28.el7_5.1                                     
  mpfr.x86_64 0:3.1.1-4.el7                                                     

Complete!
pywbem_os_setup.sh: Installing package: swig
Loaded plugins: product-id, search-disabled-repos, subscription-manager
Resolving Dependencies
--> Running transaction check
---> Package swig.x86_64 0:2.0.10-5.el7 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package     Arch          Version              Repository                 Size
================================================================================
Installing:
 swig        x86_64        2.0.10-5.el7         rhel-7-server-rpms        1.3 M

Transaction Summary
================================================================================
Install  1 Package

Total download size: 1.3 M
Installed size: 4.7 M
Downloading packages:
swig-2.0.10-5.el7.x86_64.rpm                               | 1.3 MB   00:01     
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : swig-2.0.10-5.el7.x86_64                                     1/1 
  Verifying  : swig-2.0.10-5.el7.x86_64                                     1/1 

Installed:
  swig.x86_64 0:2.0.10-5.el7                                                    

Complete!
pywbem_os_setup.sh: Installing package: python-devel
Loaded plugins: product-id, search-disabled-repos, subscription-manager
Package python-devel-2.7.5-68.el7.x86_64 already installed and latest version
Nothing to do
(pywbem27)kschopmeyer@smipypingtest3 pywbem [master] $ make clobber build install check test
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
makefile: Removing temporary build products
find . -name "*.pyc" -delete
sh -c "find . -name \"__pycache__\" |xargs rm -Rf __pycache__"
rm -Rf tmp_ tmp_*
rm -f MANIFEST parser.out .coverage pywbem/parser.out test_27.tmp.log
rm -Rf build tmp_install testtmp testsuite/testtmp .cache pywbem.egg-info .eggs
makefile: Done removing temporary build products
makefile: Target clean done.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
makefile: Removing everything for a fresh start
rm -f *.done pylint.log flake8.log epydoc.log test_*.log pywbem/mofparsetab.py pywbem/moflextab.py dist/pywbem--py2.py3-none-any.whl dist/pywbem-.tar.gz pywbem/*,cover wbemcli.log
rm -Rf build_doc .tox coverage_html
makefile: Done removing everything for a fresh start
makefile: Target clobber done.
Error: Package version could not be determined (requires pbr; run "make install")
make: *** [_check_version] Error 1
(pywbem27)kschopmeyer@smipypingtest3 pywbem [master] $ make build install check test
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Error: Package version could not be determined (requires pbr; run "make install")
make: *** [_check_version] Error 1
(pywbem27)kschopmeyer@smipypingtest3 pywbem [master] $ make install build
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pbr.version
makefile: Installing OS-level installation and runtime requirements
./pywbem_os_setup.sh install
pywbem_os_setup.sh: Installing OS-level prerequisite packages for install on platform Linux (distro: rhel, family: redhat)...
pywbem_os_setup.sh: Using installer: yum
Loaded plugins: product-id, search-disabled-repos, subscription-manager
rhel-7-server-rpms                                       | 3.5 kB     00:00     
rhel-server-rhscl-7-rpms                                 | 3.5 kB     00:00     
Metadata Cache Created
pywbem_os_setup.sh: Installing package: openssl-devel
Loaded plugins: product-id, search-disabled-repos, subscription-manager
Package 1:openssl-devel-1.0.2k-12.el7.x86_64 already installed and latest version
Nothing to do
pywbem_os_setup.sh: Installing package: gcc-c++
Loaded plugins: product-id, search-disabled-repos, subscription-manager
Package gcc-c++-4.8.5-28.el7_5.1.x86_64 already installed and latest version
Nothing to do
pywbem_os_setup.sh: Installing package: swig
Loaded plugins: product-id, search-disabled-repos, subscription-manager
Package swig-2.0.10-5.el7.x86_64 already installed and latest version
Nothing to do
pywbem_os_setup.sh: Installing package: python-devel
Loaded plugins: product-id, search-disabled-repos, subscription-manager
Package python-devel-2.7.5-68.el7.x86_64 already installed and latest version
Nothing to do
touch install_os.done
makefile: Done installing OS-level installation and runtime requirements
makefile: Installing/upgrading basic Python packages (pip, etc., with PACKAGE_LEVEL=latest)
python remove_duplicate_setuptools.py
Removing duplicate metadata directories of package: setuptools
Current Python version: 2.7
Version of imported setuptools package: 18.0.1
Site packages directory of imported package: /home/kschopmeyer/.virtualenvs/pywbem27/lib/python2.7/site-packages
Found matching metadata directory: /home/kschopmeyer/.virtualenvs/pywbem27/lib/python2.7/site-packages/setuptools-18.0.1.dist-info
python -m pip install --upgrade pip setuptools wheel
You are using pip version 7.1.0, however version 10.0.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Collecting pip
  Downloading https://files.pythonhosted.org/packages/0f/74/ecd13431bcc456ed390b44c8a6e917c1820365cbebcb6a8974d1cd045ab4/pip-10.0.1-py2.py3-none-any.whl (1.3MB)
    100% |████████████████████████████████| 1.3MB 182kB/s 
Collecting setuptools
  Downloading https://files.pythonhosted.org/packages/7f/e1/820d941153923aac1d49d7fc37e17b6e73bfbd2904959fffbad77900cf92/setuptools-39.2.0-py2.py3-none-any.whl (567kB)
    100% |████████████████████████████████| 569kB 385kB/s 
Collecting wheel
  Downloading https://files.pythonhosted.org/packages/81/30/e935244ca6165187ae8be876b6316ae201b71485538ffac1d718843025a9/wheel-0.31.1-py2.py3-none-any.whl (41kB)
    100% |████████████████████████████████| 45kB 2.4MB/s 
Installing collected packages: pip, setuptools, wheel
  Found existing installation: pip 7.1.0
    Uninstalling pip-7.1.0:
      Successfully uninstalled pip-7.1.0
  Found existing installation: setuptools 18.0.1
    Uninstalling setuptools-18.0.1:
      Successfully uninstalled setuptools-18.0.1
  Found existing installation: wheel 0.24.0
    Uninstalling wheel-0.24.0:
      Successfully uninstalled wheel-0.24.0
Successfully installed pip-10.0.1 setuptools-39.2.0 wheel-0.31.1
pip install --upgrade pbr
Collecting pbr
  Downloading https://files.pythonhosted.org/packages/b3/5d/c196041ffdf3e34ba206db6d61d1f893a75e1f3435699ade9bd65e089a3d/pbr-4.0.4-py2.py3-none-any.whl (98kB)
    100% |████████████████████████████████| 102kB 1.1MB/s 
Installing collected packages: pbr
Successfully installed pbr-4.0.4
touch install_basic.done
makefile: Done installing/upgrading basic Python packages
makefile: Installing pywbem (editable) and its Python runtime prerequisites (with PACKAGE_LEVEL=latest)
rm -Rf build pywbem.egg-info .eggs
rm -f PKG-INFO
pip install --upgrade -r requirements.txt
Ignoring ordereddict: markers 'python_version == "2.6"' don't match your environment
Requirement already up-to-date: pbr>=1.10.0 in /home/kschopmeyer/.virtualenvs/pywbem27/lib/python2.7/site-packages (from -r requirements.txt (line 12)) (4.0.4)
Collecting six>=1.10.0 (from -r requirements.txt (line 13))
  Downloading https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl
Collecting ply>=3.10 (from -r requirements.txt (line 14))
  Downloading https://files.pythonhosted.org/packages/a3/58/35da89ee790598a0700ea49b2a66594140f44dec458c07e8e3d4979137fc/ply-3.11-py2.py3-none-any.whl (49kB)
    100% |████████████████████████████████| 51kB 1.3MB/s 
Collecting PyYAML>=3.12 (from -r requirements.txt (line 15))
  Downloading https://files.pythonhosted.org/packages/4a/85/db5a2df477072b2902b0eb892feb37d88ac635d36245a72a6a69b23b383a/PyYAML-3.12.tar.gz (253kB)
    100% |████████████████████████████████| 256kB 2.3MB/s 
Collecting M2Crypto>=0.30.1 (from -r requirements.txt (line 17))
  Downloading https://files.pythonhosted.org/packages/41/50/7d85dc99b1c4f29eca83873d851ec29a8e484a66b31351e62e30be9db7d1/M2Crypto-0.30.1.tar.gz (1.3MB)
    100% |████████████████████████████████| 1.3MB 2.1MB/s 
Collecting mock>=2.0.0 (from -r requirements.txt (line 19))
  Downloading https://files.pythonhosted.org/packages/e6/35/f187bdf23be87092bd0f1200d43d23076cee4d0dec109f195173fd3ebc79/mock-2.0.0-py2.py3-none-any.whl (56kB)
    100% |████████████████████████████████| 61kB 2.9MB/s 
Collecting typing (from M2Crypto>=0.30.1->-r requirements.txt (line 17))
  Downloading https://files.pythonhosted.org/packages/0d/4d/4e5985d075d241d686a1663fa1f88b61d544658d08c1375c7c6aac32afc3/typing-3.6.4-py2-none-any.whl
Collecting funcsigs>=1; python_version < "3.3" (from mock>=2.0.0->-r requirements.txt (line 19))
  Downloading https://files.pythonhosted.org/packages/69/cb/f5be453359271714c01b9bd06126eaf2e368f1fddfff30818754b5ac2328/funcsigs-1.0.2-py2.py3-none-any.whl
Building wheels for collected packages: PyYAML, M2Crypto
  Running setup.py bdist_wheel for PyYAML ... done
  Stored in directory: /home/kschopmeyer/.cache/pip/wheels/03/05/65/bdc14f2c6e09e82ae3e0f13d021e1b6b2481437ea2f207df3f
  Running setup.py bdist_wheel for M2Crypto ... done
  Stored in directory: /home/kschopmeyer/.cache/pip/wheels/a8/e3/71/ae09232b6b31347cd5913364ad4647980b46ef590abb252498
Successfully built PyYAML M2Crypto
Installing collected packages: six, ply, PyYAML, typing, M2Crypto, funcsigs, mock
Successfully installed M2Crypto-0.30.1 PyYAML-3.12 funcsigs-1.0.2 mock-2.0.0 ply-3.11 six-1.11.0 typing-3.6.4
pip install --upgrade -e .
Obtaining file:///home/kschopmeyer/python_work/pywbem
Requirement not upgraded as not directly required: pbr>=1.10.0 in /home/kschopmeyer/.virtualenvs/pywbem27/lib/python2.7/site-packages (from pywbem==0.12.1.dev45) (4.0.4)
Requirement not upgraded as not directly required: six>=1.10.0 in /home/kschopmeyer/.virtualenvs/pywbem27/lib/python2.7/site-packages (from pywbem==0.12.1.dev45) (1.11.0)
Requirement not upgraded as not directly required: ply>=3.10 in /home/kschopmeyer/.virtualenvs/pywbem27/lib/python2.7/site-packages (from pywbem==0.12.1.dev45) (3.11)
Requirement not upgraded as not directly required: PyYAML>=3.12 in /home/kschopmeyer/.virtualenvs/pywbem27/lib/python2.7/site-packages (from pywbem==0.12.1.dev45) (3.12)
Requirement not upgraded as not directly required: M2Crypto>=0.30.1 in /home/kschopmeyer/.virtualenvs/pywbem27/lib/python2.7/site-packages (from pywbem==0.12.1.dev45) (0.30.1)
Requirement not upgraded as not directly required: mock>=2.0.0 in /home/kschopmeyer/.virtualenvs/pywbem27/lib/python2.7/site-packages (from pywbem==0.12.1.dev45) (2.0.0)
Requirement not upgraded as not directly required: typing in /home/kschopmeyer/.virtualenvs/pywbem27/lib/python2.7/site-packages (from M2Crypto>=0.30.1->pywbem==0.12.1.dev45) (3.6.4)
Requirement not upgraded as not directly required: funcsigs>=1; python_version < "3.3" in /home/kschopmeyer/.virtualenvs/pywbem27/lib/python2.7/site-packages (from mock>=2.0.0->pywbem==0.12.1.dev45) (1.0.2)
Installing collected packages: pywbem
  Running setup.py develop for pywbem
Successfully installed pywbem
cp -r pywbem.egg-info/PKG-INFO .
touch install_pywbem.done
makefile: Done installing pywbem and its Python runtime prerequisites
python -c "import pywbem; print('ok, version=%r'%pywbem.__version__)"
ok, version='0.12.1.dev45'
python -c "import pywbem_mock; print('ok')"
ok
touch install.done
makefile: Target install done.
Error: Package version could not be determined (requires pbr; run "make install")
make: *** [_check_version] Error 1
(pywbem27)kschopmeyer@smipypingtest3 pywbem [master] $ make install
makefile: Target install done.
(pywbem27)kschopmeyer@smipypingtest3 pywbem [master] $ pip list
Package    Version      Location                            
---------- ------------ ------------------------------------
distro     1.3.0        
funcsigs   1.0.2        
M2Crypto   0.30.1       
mock       2.0.0        
pbr        4.0.4        
pip        10.0.1       
ply        3.11         
pywbem     0.12.1.dev45 /home/kschopmeyer/python_work/pywbem
PyYAML     3.12         
setuptools 39.2.0       
six        1.11.0       
typing     3.6.4        
wheel      0.31.1       
(pywbem27)kschopmeyer@smipypingtest3 pywbem [master] $ ./pywbem_os_setup.sh

--- a/PKG-INFO
+++ b/PKG-INFO
@@ -1,6 +1,6 @@
 Metadata-Version: 1.2
 Name: pywbem
-Version: 0.12.1.dev39
+Version: 0.12.1.dev46
 Summary: pywbem - A WBEM client
 Home-page: http://pywbem.github.io/pywbem/
 Author: Tim Potter

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -107,7 +107,7 @@ install:
   # See GenShim issue https://github.com/chocolatey/shimgen/issues/43
   # Therefore, we install swig here in order to create the swig.exe shim file.
   # TODO: Remove the circumvention once it works in pywbem_os_setup.bat.
-  - choco install -r -y swig
+  - choco install -r -y --allow-empty-checksums swig
   - where swig
 
   # Examine the final environment

--- a/docs/changes.rst
+++ b/docs/changes.rst
@@ -147,14 +147,15 @@ Released: not yet
   the `M2CryptoWin32/64` packages. For details, see the installation section
   in the documentation.
 
-
 * Made exception messages more explicit in the ValueMapping and WBEMServer
   classes. Issue #1281.
  
-
 * Docs: Added a shell command for determining the version of an installed
   pywbem package, that covers all released pywbem versions (Issue #1246).
 
+* Make: Eliminated the confusing but unproblematic error message about
+  pbr importing when running certain make targets in a freshly created
+  Python environment. Issue #1288.
 
 **Cleanup**
 

--- a/makefile
+++ b/makefile
@@ -77,7 +77,7 @@ coverage_html_dir := coverage_html
 # e.g. because the pywbem.egg-info directory or the PKG-INFO file are deleted,
 # when a new version tag has been assigned. Therefore, this variable is assigned with
 # "=" so that it is evaluated every time it is used.
-package_version = $(shell $(PYTHON_CMD) -c "from pbr.version import VersionInfo; print(VersionInfo('pywbem').release_string())")
+package_version := $(shell $(PYTHON_CMD) -c $$'try:\n from pbr.version import VersionInfo\nexcept ImportError:\n pass\nelse:\n print(VersionInfo("$(package_name)").release_string())\n')
 
 # Python versions
 python_version := $(shell $(PYTHON_CMD) -c "import sys; sys.stdout.write('%s.%s.%s'%sys.version_info[0:3])")
@@ -88,10 +88,10 @@ dist_dir := dist
 
 # Distribution archives
 # These variables are set with "=" for the same reason as package_version.
-bdist_file = $(dist_dir)/$(package_name)-$(package_version)-py2.py3-none-any.whl
-sdist_file = $(dist_dir)/$(package_name)-$(package_version).tar.gz
+bdist_file := $(dist_dir)/$(package_name)-$(package_version)-py2.py3-none-any.whl
+sdist_file := $(dist_dir)/$(package_name)-$(package_version).tar.gz
 
-dist_files = $(bdist_file) $(sdist_file)
+dist_files := $(bdist_file) $(sdist_file)
 
 # Source files in the packages
 package_py_files := \
@@ -319,7 +319,7 @@ develop.done: install.done develop_os.done install_basic.done dev-requirements.t
 	@echo "makefile: Done installing Python development requirements"
 
 .PHONY: build
-build: _check_version $(bdist_file) $(sdist_file)
+build: $(bdist_file) $(sdist_file)
 	@echo "makefile: Target $@ done."
 
 .PHONY: builddoc
@@ -459,7 +459,7 @@ MANIFEST.in: makefile
 # regenerate MANIFEST. Otherwise, changes in MANIFEST.in will not be used.
 # Note: Deleting build is a safeguard against picking up partial build products
 # which can lead to incorrect hashbangs in the pywbem scripts in wheel archives.
-$(bdist_file) $(sdist_file): setup.py MANIFEST.in $(dist_dependent_files) $(moftab_files)
+$(bdist_file) $(sdist_file): _check_version setup.py MANIFEST.in $(dist_dependent_files) $(moftab_files)
 	@echo "makefile: Creating the distribution archive files"
 	rm -rf MANIFEST $(package_name).egg-info .eggs build
 	rm -f PKG-INFO

